docs/cpp/evolution.md

```diff
@@ -0,0 +1,244 @@
+# Schema Evolution
+
+Data models often change over time, thus, yardl provides support for evolving your schema in a cross-version compatible manner.
+
+When deploying protocol readers and writers generated from the current version of your schema, you may need the ability to read and write older versions of your data.
+Conversely, you may want to update older versions of your software to read and write the current version.
+
+Yardl supports cross-version compatibility by *statically* generating conversions to and from other versions of your schema (i.e. at yardl codegen time).
+
+In the future, yardl will support *dynamic* compatibility between schema versions, enabling backward- and forward-compatibility between any schema version at runtime.
```

Comment:
We should also call out that for the moment, we only support this for the binary encoding format using C++.